cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -720,50 +730,121 @@ function getPriceHtml(product) {
+                displayPrice = computedPrice; 
+            } else if (computedPrice > displayPrice) {
+                // If pricebooks show a higher original price
+                strikeoutPrice = computedPrice;
+            }
+        }
     }
 
-    if (promotions && promotions[algoliaData.currencyCode]) {
-        var productPromos = promotions[algoliaData.currencyCode];
-        var minPrice = algoliaData.recordModel === 'master-level' ? (variant.price[algoliaData.currencyCode] ? variant.price[algoliaData.currencyCode] : variant.price) : item.price;
-        for (var i = 0; i < activeCustomerPromotions.length; i++) {
-            for (var j = 0; j < productPromos.length; j++) {
-                if (productPromos[j].promoId === activeCustomerPromotions[i].id && productPromos[j].price < minPrice) {
-                    minPrice = productPromos[j].price;
-                    calloutMsg = activeCustomerPromotions[i].calloutMsg;
-                }
-            }
+    if (item.promotionalPrice && item.promotionalPrice < item.price) {
+        strikeoutPrice = item.price;
+        displayPrice = item.promotionalPrice;
+    }
+
+    if (item.promotions && item.promotions[algoliaData.currencyCode]) {
+        const { promotionalPriceFromPromos, msg } = applyPromotions(item, algoliaData, activeCustomerPromos);
+        if (promotionalPriceFromPromos < displayPrice) {
+            displayPrice = promotionalPriceFromPromos;
+            calloutMsg = msg;
         }
+    }
+
+    if (!strikeoutPrice) {
+        strikeoutPrice = item.price > displayPrice ? item.price : null;
+    }
 
-        return {
-            price: minPrice,
-            calloutMsg: calloutMsg,
+    return { displayPrice, strikeoutPrice, calloutMsg };
+}
+
+/**
+ * Apply promotions to find a lower promotional price if available
+ * @param {Object} item The item
+ * @param {Object} algoliaData The Algolia data config
+ * @param {Array} activeCustomerPromotions Active promotions
+ * @returns {Object} { promotionalPriceFromPromos, msg }
+ */
+function applyPromotions(item, algoliaData, activeCustomerPromos) {
+    let promotionalPriceFromPromos = item.price;
+    let msg = '';
+
+    const productPromos = item.promotions[algoliaData.currencyCode];
+    for (var i = 0; i < activeCustomerPromos.length; i++) {
+        for (var j = 0; j < productPromos.length; j++) {
+            if (productPromos[j].promoId === activeCustomerPromos[i].id && productPromos[j].price < promotionalPriceFromPromos) {
+                promotionalPriceFromPromos = productPromos[j].price;
+                msg = activeCustomerPromos[i].calloutMsg;
+            }
         }
     }
+    return { promotionalPriceFromPromos, msg };
+}
 
-    if (variant.promotionalPrice) {
-        return {
-            price: variant.promotionalPrice[algoliaData.currencyCode] || variant.promotionalPrice,
-            calloutMsg: '',
+/**
+ * Apply pricebooks to adjust prices if no promotional price was found
```

Comment:
This function only focus on pricebooks, it doesn't take into account promotional prices. The comment needs to be updated.